lib/tesla/middleware/timeout.ex

```diff
@@ -23,18 +24,29 @@ defmodule Tesla.Middleware.Timeout do
   }
 
   @default_timeout 1_000
+  @default_shutdown true
 
   def call(env, next, opts) do
     opts    = opts || []
     timeout   = Keyword.get(opts, :timeout, @default_timeout)
+    shutdown  = Keyword.get(opts, :shutdown, @default_shutdown)
 
     task = safe_async(fn -> Tesla.run(env, next) end)
     try do
       task
       |> Task.await(timeout)
       |> repass_error
     catch :exit, {:timeout, _} ->
-      Process.exit(task.pid, :kill)
+      if shutdown do
+          if Version.match?(System.version(), ">= 1.4.0") do
+            Task.shutdown(task, 0)
+          else
+            Process.unlink(task.pid)
+            Process.exit(task.pid, :kill)
+          end
+      else
+          Process.exit(task.pid, :kill)
```

Comment:
I think you can remove this and whole outer `if`.
In original PR (#119) `Task.shutdown` was replaced by `Process.exit(:kill)` to maintain compatibility with Elixir 1.3. Inner `if` looks like correct solution.

Reply:
ok